presto-main/src/test/java/com/facebook/presto/sql/planner/TestRowExpressionFormatter.java

```diff
@@ -116,11 +143,74 @@ public void testConstants()
     }
 
+    @Test
+    public void testCalls()
+    {
+        // setup
+        RowExpression callExpression;
+
+        // arithmetic
+        callExpression = exampleCallExpression(ADD);
+        assertEquals(format(callExpression), "(#1) + (BIGINT 5)");
+        callExpression = exampleCallExpression(SUBTRACT);
+        assertEquals(format(callExpression), "(#1) - (BIGINT 5)");
+        callExpression = exampleCallExpression(MULTIPLY);
+        assertEquals(format(callExpression), "(#1) * (BIGINT 5)");
+        callExpression = exampleCallExpression(DIVIDE);
+        assertEquals(format(callExpression), "(#1) / (BIGINT 5)");
+        callExpression = exampleCallExpression(MODULUS);
+        assertEquals(format(callExpression), "(#1) % (BIGINT 5)");
+
+        // comparison
+        callExpression = exampleCallExpression(GREATER_THAN);
+        assertEquals(format(callExpression), "(#1) > (BIGINT 5)");
+        callExpression = exampleCallExpression(LESS_THAN);
+        assertEquals(format(callExpression), "(#1) < (BIGINT 5)");
+        callExpression = exampleCallExpression(GREATER_THAN_OR_EQUAL);
+        assertEquals(format(callExpression), "(#1) >= (BIGINT 5)");
+        callExpression = exampleCallExpression(LESS_THAN_OR_EQUAL);
+        assertEquals(format(callExpression), "(#1) <= (BIGINT 5)");
+        callExpression = exampleCallExpression(EQUAL);
+        assertEquals(format(callExpression), "(#1) = (BIGINT 5)");
+        callExpression = exampleCallExpression(NOT_EQUAL);
+        assertEquals(format(callExpression), "(#1) <> (BIGINT 5)");
+        callExpression = exampleCallExpression(IS_DISTINCT_FROM);
+        assertEquals(format(callExpression), "(#1) IS DISTINCT FROM (BIGINT 5)");
+
+        // nested expressions
+        RowExpression temp = exampleCallExpression(ADD);
+        callExpression = call(
+                SUBTRACT.name(),
+                functionManager.resolveOperator(SUBTRACT, fromTypes(BIGINT, BIGINT)),
+                BIGINT,
+                field(4, BIGINT),
+                temp);
+        assertEquals(format(callExpression), "(#4) - ((#1) + (BIGINT 5))");
+
+        // other functions
+        callExpression = call(
+            CAST.name(),
+            functionManager.lookupCast(CastType.CAST, TINYINT.getTypeSignature(), BIGINT.getTypeSignature()),
+            BIGINT,
+            constant(1, TINYINT));
+        assertEquals(format(callExpression), "CAST(TINYINT 1)");
+    }
+
     protected static Object decimal(String decimalString)
     {
         return Decimals.parseIncludeLeadingZerosInPrecision(decimalString).getObject();
     }
 
+    private static CallExpression exampleCallExpression(OperatorType type)
+    {
+        return call(
+                type.name(),
+                functionManager.resolveOperator(type, fromTypes(BIGINT, BIGINT)),
+                BIGINT,
+                field(1, BIGINT),
```

Comment:
Maybe use `VariableReferenceExpression`. The `field` is used in a later phase in a plan (which you don't have to worry about the difference for now).